### ISSUE TITLE
Blinking cursor

### ISSUE THREAD

**Issue by hermanbergwerf** [daniel3735928559/guppy#29]
I think a blinking cursor gives nice feedback to users about where their cursor is. It might be very easy to implement this using CSS3 (http://stackoverflow.com/questions/16344354/how-to-make-blinking-flashing-text-with-css3). It would also be nice if the cursor is a little bit thicker and higher, I played around bit with  the `Guppy.kb.CONSTANTS`, and some CSS rules, but it seems the current architecture doesn't allow much tweaking. If there are some extra CSS class selectors (e.g. a `.cursor-active`) I could add some custom styles (such as thickness, I'm okay with a cursor that displaces text, and I don't really like the negative margin to compensate). In fact a CSS3 transforms might even enable a cursor that has a fairly usual height without breaking anything.

P.S. I really like your project! I'm using the Dart language, and I'll try to create a Dart wrapper for this. The code seems very small so I could, theoretically, even write a Dart port! (Dart is really nice, much cleaner than JavaScript). I'm unfamiliar with the XSL stuff you are using, it seems a pretty uncommon technique (but powerful apparently).

**Comment by hermanbergwerf** [comment]
I tried this and it is fairly easy, just copy the contents from the StackOverflow thing.

Also, I found adding these rules to `.cursor` makes it display a little nicer:
```css
margin-right: 0 !important;
border-width: 2px !important;
transform: scaleY(1.4);
```

You can also just use the `transform` to get a slightly larger cursor.

**Comment by daniel3735928559** [comment]
I agree about the blinking cursor.  One slight issue is that there are few different cursor objects--the main one in red, but also the "start of selection" cursor in blue, and the phantom "this is where your cursor would be if you clicked right now" cursor in grey.  I don't think it would be normal for all of these to blink, so there would have to be a way of differentiating them in CSS.  

Right now, the cursor is implemented using a custom `\cursor` KaTeX command that I added, which supports a TeX argument for height and an optional one for vertical offset, but not one for width.  It also gives the cursor a `cursor` CSS class, so potentially you could get some of the desired behaviour by adding CSS rules for that class.  

One could imagine instead a world in which the cursor was simply `\class{cursor}{\rule{...}}` (`\class` being the other custom KaTeX command I [added](https://github.com/Khan/KaTeX/pull/507), but which seems less controversial), and then all styling of the various cursors would be done only by CSS rules.  This would make the blinking cursor pretty easy, as well as the modifications you want. (I personally don't particularly like how bits of the equation shift around as I navigate when the cursor does have nonzero effective width, but if it's just modifying CSS, then anyone can do what they find looks best.)

Glad you like it!  I am not familiar with the Dart language.  My only caution about porting would be that I have a growing wishlist of further features that I hope to implement in the near-ish future (once school lets up a bit), however that influences your decision.  (I am however open to an argument that future development should be done in different language anyways.)

And lastly, in case it is useful: a word about what's going on with the XSLT business.  The pattern is that at any time, the state of the program is contained in one object that is easy to transform to anything else, so that when it comes time to, say, render, or convert to a format that other software consumes, that object can then be transformed to whatever is needed.  (So, for example, saving is just serialising that one state object).  My particular habit has been to make this one object an XML document, and to do the transformations with XSLT.  Some advantages are: XML is sort of human-readable, XSL transformations are declarative and use templates (so are pretty nice to write), XSL transformations are supported by most browsers, and updating old saved documents to an improved XML format is especially easy with a one-off XSL transform.  Some disadvantages are that XSL is ugly, and that constantly searching and updating an XML document isn't always the most efficient thing.

**Comment by daniel3735928559** [comment]
It turns out because of all the interactions between different CSS classes in KaTeX, the `\cursor` element is still needed.  I gave the main cursor an extra CSS class `.main_cursor` that can be used to add blinkyness.  Because I do like the blinking behaviour, I added it on the demo page.  (This should also let you add CSS classes (or override the default ones) if you want a differently sized cursor.)
